scalding-core/src/main/scala/com/twitter/scalding/estimation/memory/MemoryHistoryEstimator.scala

```diff
@@ -0,0 +1,117 @@
+package com.twitter.scalding.estimation.memory
+
+import com.twitter.scalding.estimation.{ FlowStepHistory, FlowStrategyInfo, HistoryEstimator, Task }
+import org.apache.hadoop.mapred.JobConf
+import org.slf4j.LoggerFactory
+
+// Tuple(MapMemory in MB for java process and container, ReduceMemory in MB for java process and container),
+// or None to keep the default.
+case class MemoryEstimate(mapMemoryInMB: Option[(Long, Long)], reduceMemoryInMB: Option[(Long, Long)])
+
+object MemoryHistoryEstimator {
+  val CommittedHeapBytes = "COMMITTED_HEAP_BYTES"
+  val CpuMs = "CPU_MILLISECONDS"
+  val PhysicalMemoryBytes = "PHYSICAL_MEMORY_BYTES"
+  val GCTimeMs = "GC_TIME_MILLIS"
+
+  implicit class MemoryRichTask(val task: Task) extends AnyVal {
+    def committedHeapBytes: Option[Long] = task.counters.get(CommittedHeapBytes)
+  }
+}
+
+trait MemoryHistoryEstimator extends HistoryEstimator[MemoryEstimate] {
```

Comment:
What's the convention for naming default implementations? I think we should make it obvious that this is a specific algorithm for estimating the next memory allocations and not the only one.